src/notebook/widget.ts

```diff
@@ -201,6 +200,7 @@ class NotebookWidget extends Widget {
       widget = new BaseCellWidget(cell);
     }
     widget.addClass(NB_CELL_CLASS);
+    widget.input.editor.addClass(NB_EDITOR_CLASS);
```

Comment:
I guess this means that the cell widget can't swap out the editor widget under our noses. Something to keep in mind...